src/structure/4C_structure_timint_impl.cpp

```diff
@@ -3171,6 +3171,40 @@ void Solid::TimIntImpl::cmt_linear_solve()
       cmtbridge_->get_strategy().build_saddle_point_system(
           stiff_, fres_, disi_, dbcmaps_, blockMat, blocksol, blockrhs);
 
+      // compute the nullspace vectors for the Lagrange multiplier field for MueLu
+      if (contactsolver_->params().isSublist("Belos Parameters"))
+        if (contactsolver_->params()
+                .sublist("Belos Parameters")
+                .get<std::string>("Preconditioner Type") == "ML")
+        {
+          int dim_nullspace = contactsolver_->params()
+                                  .sublist("Inverse2")
+                                  .sublist("MueLu Parameters")
+                                  .get<int>("PDE equations", -1);
+
+          // get the degree of freedom map from the block matrix
+          Epetra_Operator* raw_block_mat = blockMat.get();
+          std::shared_ptr<Core::LinAlg::BlockSparseMatrixBase> block_mat_blocked_operator =
+              std::dynamic_pointer_cast<Core::LinAlg::BlockSparseMatrixBase>(
+                  std::shared_ptr<Epetra_Operator>(raw_block_mat, [](Epetra_Operator*) {}));
```

Comment:
```suggestion
                  Core::Utils::shared_ptr_from_ref(*raw_block_mat,));
```

This at least makes this crime available to textual search ;)